IllegalStateException: Duplicate Key

When using the emojis.json found [here](https://github.com/felldo/JEmoji/blob/master/lib/src/main/resources/emojis.json), there's a duplicate key error when it tries to map the `EMOJI_UNICODE_TO_EMOJI` map. 

This is the full stacktrace:

```
Caused by: java.lang.IllegalStateException: Duplicate key �? (attempted merging values Emoji{emoji='�?', unicode='?', discordAliases=[:grin:], githubAliases=[:grin:], slackAliases=[:grin:], hasFitzpatrick=false, hasHairStyle=false, version=0.6, qualification=FULLY_QUALIFIED, description='beaming face with smiling eyes', group=SMILEYS_AND_EMOTION, subgroup=FACE_SMILING} and Emoji{emoji='�?', unicode='?', discordAliases=[:heart_eyes:], githubAliases=[:heart_eyes:], slackAliases=[:heart_eyes:], hasFitzpatrick=false, hasHairStyle=false, version=0.6, qualification=FULLY_QUALIFIED, description='smiling face with heart-eyes', group=SMILEYS_AND_EMOTION, subgroup=FACE_AFFECTION})
```

When I go into the debugger, it's pretty easy to tell that when the JSON file is read, the `emoji` field for both 😁 and 😍 are the same, as seen in the image below.
 
![image](https://github.com/felldo/JEmoji/assets/42523509/d823de0c-1a85-433a-a77d-05b9e2562952)

 Perhaps mapping with the Unicode instead of Emoji would fix this?

--- a/lib/build.gradle.kts
+++ b/lib/build.gradle.kts
@@ -125,74 +125,74 @@ tasks.register("copyJarToProject") {
     }
 }
 
-//publishing {
-//    publications {
-//        create<MavenPublication>("JEMOJI") {
-//            artifactId = "jemoji"
-//            from(components["java"])
-//            versionMapping {
-//                usage("java-api") {
-//                    fromResolutionOf("runtimeClasspath")
-//                }
-//                usage("java-runtime") {
-//                    fromResolutionResult()
-//                }
-//            }
-//            pom {
-//                name.set("Java Emoji")
-//                description.set(rootProject.description)
-//                url.set("https://github.com/felldo/jemoji")
-//
-//                licenses {
-//                    license {
-//                        name.set("The Apache License, Version 2.0")
-//                        url.set("https://www.apache.org/licenses/LICENSE-2.0.txt")
-//                    }
-//                }
-//
-//                inceptionYear.set("2023")
-//                developers {
-//                    developer {
-//                        id.set("felldo")
-//                        name.set("Dominic Fellbaum")
-//                        email.set("d.fellbaum@hotmail.de")
-//                        url.set("https://github.com/felldo")
-//                        timezone.set("Europe/Berlin")
-//                    }
-//                }
-//                scm {
-//                    connection.set("scm:git:https://github.com/felldo/jemoji.git")
-//                    developerConnection.set("scm:git:git@github.com:felldo/jemoji.git")
-//                    url.set("https://github.com/felldo/jemoji")
-//                }
-//            }
-//        }
-//    }
-//    repositories {
-//        maven {
-//            val isReleaseVersion = !version.toString().endsWith("SNAPSHOT")
-//            name = "OSSRH"
-//            url = if (isReleaseVersion) {
-//                uri("https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/")
-//            } else {
-//                uri("https://s01.oss.sonatype.org/content/repositories/snapshots/")
-//            }
-//            credentials {
-//                username = findProperty("NEXUS_USERNAME") as String
-//                password = findProperty("NEXUS_PASSWORD") as String
-//            }
-//        }
-//    }
-//}
-//
-//signing {
-//    val signingKey = findProperty("JEMOJI_SINGING_SECRET_KEY_RING_FILE") as String
-//    val signingKeyId = findProperty("JEMOJI_SIGNING_KEY_ID") as String
-//    val signingPassword = findProperty("JEMOJI_SIGNING_PASSWORD") as String
-//
-//    useInMemoryPgpKeys(signingKeyId, signingKey, signingPassword)
-//    sign(publishing.publications["JEMOJI"])
-//}
+publishing {
+    publications {
+        create<MavenPublication>("JEMOJI") {
+            artifactId = "jemoji"
+            from(components["java"])
+            versionMapping {
+                usage("java-api") {
+                    fromResolutionOf("runtimeClasspath")
+                }
+                usage("java-runtime") {
+                    fromResolutionResult()
+                }
+            }
+            pom {
+                name.set("Java Emoji")
+                description.set(rootProject.description)
+                url.set("https://github.com/felldo/jemoji")
+
+                licenses {
+                    license {
+                        name.set("The Apache License, Version 2.0")
+                        url.set("https://www.apache.org/licenses/LICENSE-2.0.txt")
+                    }
+                }
+
+                inceptionYear.set("2023")
+                developers {
+                    developer {
+                        id.set("felldo")
+                        name.set("Dominic Fellbaum")
+                        email.set("d.fellbaum@hotmail.de")
+                        url.set("https://github.com/felldo")
+                        timezone.set("Europe/Berlin")
+                    }
+                }
+                scm {
+                    connection.set("scm:git:https://github.com/felldo/jemoji.git")
+                    developerConnection.set("scm:git:git@github.com:felldo/jemoji.git")
+                    url.set("https://github.com/felldo/jemoji")
+                }
+            }
+        }
+    }
+    repositories {
+        maven {
+            val isReleaseVersion = !version.toString().endsWith("SNAPSHOT")
+            name = "OSSRH"
+            url = if (isReleaseVersion) {
+                uri("https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/")
+            } else {
+                uri("https://s01.oss.sonatype.org/content/repositories/snapshots/")
+            }
+            credentials {
+                username = findProperty("NEXUS_USERNAME") as String
+                password = findProperty("NEXUS_PASSWORD") as String
+            }
+        }
+    }
+}
+
+signing {
+    val signingKey = findProperty("JEMOJI_SINGING_SECRET_KEY_RING_FILE") as String
+    val signingKeyId = findProperty("JEMOJI_SIGNING_KEY_ID") as String
+    val signingPassword = findProperty("JEMOJI_SIGNING_PASSWORD") as String
+
+    useInMemoryPgpKeys(signingKeyId, signingKey, signingPassword)
+    sign(publishing.publications["JEMOJI"])
+}
 
 tasks.withType<Javadoc>().configureEach {
     options {

--- a/lib/build.gradle.kts
+++ b/lib/build.gradle.kts
@@ -100,6 +100,18 @@ java {
     }
 }
 
+tasks.withType<JavaCompile> {
+    options.encoding = "UTF-8"
+}
+
+tasks.withType<Test> {
+    systemProperty("file.encoding", "UTF-8")
+}
+
+tasks.withType<Javadoc>{
+    options.encoding = "UTF-8"
+}
+
 tasks.named("build") {
     finalizedBy("copyJarToProject")
 }
@@ -113,74 +125,74 @@ tasks.register("copyJarToProject") {
     }
 }
 
-publishing {
-    publications {
-        create<MavenPublication>("JEMOJI") {
-            artifactId = "jemoji"
-            from(components["java"])
-            versionMapping {
-                usage("java-api") {
-                    fromResolutionOf("runtimeClasspath")
-                }
-                usage("java-runtime") {
-                    fromResolutionResult()
-                }
-            }
-            pom {
-                name.set("Java Emoji")
-                description.set(rootProject.description)
-                url.set("https://github.com/felldo/jemoji")
-
-                licenses {
-                    license {
-                        name.set("The Apache License, Version 2.0")
-                        url.set("https://www.apache.org/licenses/LICENSE-2.0.txt")
-                    }
-                }
-
-                inceptionYear.set("2023")
-                developers {
-                    developer {
-                        id.set("felldo")
-                        name.set("Dominic Fellbaum")
-                        email.set("d.fellbaum@hotmail.de")
-                        url.set("https://github.com/felldo")
-                        timezone.set("Europe/Berlin")
-                    }
-                }
-                scm {
-                    connection.set("scm:git:https://github.com/felldo/jemoji.git")
-                    developerConnection.set("scm:git:git@github.com:felldo/jemoji.git")
-                    url.set("https://github.com/felldo/jemoji")
-                }
-            }
-        }
-    }
-    repositories {
-        maven {
-            val isReleaseVersion = !version.toString().endsWith("SNAPSHOT")
-            name = "OSSRH"
-            url = if (isReleaseVersion) {
-                uri("https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/")
-            } else {
-                uri("https://s01.oss.sonatype.org/content/repositories/snapshots/")
-            }
-            credentials {
-                username = findProperty("NEXUS_USERNAME") as String
-                password = findProperty("NEXUS_PASSWORD") as String
-            }
-        }
-    }
-}
-
-signing {
-    val signingKey = findProperty("JEMOJI_SINGING_SECRET_KEY_RING_FILE") as String
-    val signingKeyId = findProperty("JEMOJI_SIGNING_KEY_ID") as String
-    val signingPassword = findProperty("JEMOJI_SIGNING_PASSWORD") as String
-
-    useInMemoryPgpKeys(signingKeyId, signingKey, signingPassword)
-    sign(publishing.publications["JEMOJI"])
-}
+//publishing {
+//    publications {
+//        create<MavenPublication>("JEMOJI") {
+//            artifactId = "jemoji"
+//            from(components["java"])
+//            versionMapping {
+//                usage("java-api") {
+//                    fromResolutionOf("runtimeClasspath")
+//                }
+//                usage("java-runtime") {
+//                    fromResolutionResult()
+//                }
+//            }
+//            pom {
+//                name.set("Java Emoji")
+//                description.set(rootProject.description)
+//                url.set("https://github.com/felldo/jemoji")
+//
+//                licenses {
+//                    license {
+//                        name.set("The Apache License, Version 2.0")
+//                        url.set("https://www.apache.org/licenses/LICENSE-2.0.txt")
+//                    }
+//                }
+//
+//                inceptionYear.set("2023")
+//                developers {
+//                    developer {
+//                        id.set("felldo")
+//                        name.set("Dominic Fellbaum")
+//                        email.set("d.fellbaum@hotmail.de")
+//                        url.set("https://github.com/felldo")
+//                        timezone.set("Europe/Berlin")
+//                    }
+//                }
+//                scm {
+//                    connection.set("scm:git:https://github.com/felldo/jemoji.git")
+//                    developerConnection.set("scm:git:git@github.com:felldo/jemoji.git")
+//                    url.set("https://github.com/felldo/jemoji")
+//                }
+//            }
+//        }
+//    }
+//    repositories {
+//        maven {
+//            val isReleaseVersion = !version.toString().endsWith("SNAPSHOT")
+//            name = "OSSRH"
+//            url = if (isReleaseVersion) {
+//                uri("https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/")
+//            } else {
+//                uri("https://s01.oss.sonatype.org/content/repositories/snapshots/")
+//            }
+//            credentials {
+//                username = findProperty("NEXUS_USERNAME") as String
+//                password = findProperty("NEXUS_PASSWORD") as String
+//            }
+//        }
+//    }
+//}
+//
+//signing {
+//    val signingKey = findProperty("JEMOJI_SINGING_SECRET_KEY_RING_FILE") as String
+//    val signingKeyId = findProperty("JEMOJI_SIGNING_KEY_ID") as String
+//    val signingPassword = findProperty("JEMOJI_SIGNING_PASSWORD") as String
+//
+//    useInMemoryPgpKeys(signingKeyId, signingKey, signingPassword)
+//    sign(publishing.publications["JEMOJI"])
+//}
 
 tasks.withType<Javadoc>().configureEach {
     options {

